vinyldns/resource_zone.go

```diff
@@ -60,8 +60,52 @@ func resourceVinylDNSZone() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"updated": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"latest_sync": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
 			"transfer_connection": connectionSchema(),
 			"zone_connection":     connectionSchema(),
+			"acl_rule": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"access_level": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"record_mask": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"user_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"group_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"description": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "Managed by Terraform",
+						},
+						"record_types": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
```

Comment:
interesting thing here, I am not 100% sure how the record_types serializes to JSON, but looking at the API code, it defaults to `Set.empty` which means it should _always_ appear however it might be empty `[ ] `

Reply:
`record_types` allows the user to declare Terraform HCL like this, for example:
https://github.com/vinyldns/terraform-provider-vinyldns/blob/new-zone-attrs/vinyldns/resource_zone_test.go#L246

Then, the provider CRUD methods read that value and send it along in `go-vinyldns` client methods like so: https://github.com/vinyldns/terraform-provider-vinyldns/blob/new-zone-attrs/vinyldns/resource_zone.go#L446

Note that `go-vinyldns` serializes this ☝️into JSON in structs like this:
https://github.com/vinyldns/go-vinyldns/blob/0.9.4/vinyldns/zones_resources.go#L30

Also note that ACLRule.RecordTypes has no [omitempty](https://golang.org/pkg/encoding/json/); this means it will always be sent as `[]` in the event that no `record_types` have been declared in the Terraform HCL:

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.

However, other `ACLRule` attributes do `omitempty`. Are you suggesting that `go-vinyldns` should `omitempty` `ACLRule.RecordTypes`?

```hcl
...
	RecordMask  string   `json:"recordMask,omitempty"`
	RecordTypes []string `json:"recordTypes"`
...
```

Reply:
Thanks for the explanation.  I actually see that you have a test that has `recordTypes` not there and everything works.  I was more or less asking a question for my own information.

Note: you have the `omitempty` correct after reviewing that `zones_resources.go` file.  We are good.